ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/correction/java/JavaResolutionFactory.java

```diff
@@ -321,14 +321,18 @@ protected void modifyModel(IBaseModel model, IProgressMonitor monitor) throws Co
 					if (!(model instanceof IBundlePluginModelBase base))
 						return;
 					IBundle bundle = base.getBundleModel().getBundle();
-					String pkgId = ((ExportPackageDescription) getChangeObject()).getName();
+					ExportPackageDescription desc = (ExportPackageDescription) getChangeObject();
+					String pkgId = desc.getName();
 					IManifestHeader header = bundle.getManifestHeader(Constants.IMPORT_PACKAGE);
 					if (header == null) {
-						bundle.setHeader(Constants.IMPORT_PACKAGE, pkgId);
-					} else if (header instanceof ImportPackageHeader ipHeader) {
+						header = bundle.getModel().getFactory().createHeader(Constants.IMPORT_PACKAGE, pkgId);
+					}
+					if (header instanceof ImportPackageHeader ipHeader) {
 						int manifestVersion = BundlePluginBase.getBundleManifestVersion(bundle);
-						String versionAttr = (manifestVersion < 2) ? ICoreConstants.PACKAGE_SPECIFICATION_VERSION : Constants.VERSION_ATTRIBUTE;
-						ImportPackageObject impObject = new ImportPackageObject((ManifestHeader) header, (ExportPackageDescription) getChangeObject(), versionAttr);
+						String versionAttr = (manifestVersion < 2) ? ICoreConstants.PACKAGE_SPECIFICATION_VERSION
+								: Constants.VERSION_ATTRIBUTE;
+						ImportPackageObject impObject = new ImportPackageObject((ManifestHeader) header, desc,
+								versionAttr);
```

Comment:
If you inline `manifestVersion` and replace header by `ipHeader` we would get some nicely formatted code.
```suggestion
						String versionAttr = (BundlePluginBase.getBundleManifestVersion(bundle) < 2)
								? ICoreConstants.PACKAGE_SPECIFICATION_VERSION
								: Constants.VERSION_ATTRIBUTE;
						ImportPackageObject impObject = new ImportPackageObject(ipHeader, desc, versionAttr);
```